crates/matrix-sdk/src/event_cache/mod.rs

```diff
@@ -396,6 +404,49 @@ impl EventCacheInner {
                 continue;
             };
 
+            // Handle and cache events and relations
+            let mut relations = self.relations.write().await;
+            for ev in &joined_room_update.timeline.events {
+                if let Ok(AnySyncTimelineEvent::MessageLike(ev)) = ev.event.deserialize() {
+                    if let Some(AnyMessageLikeEventContent::RoomMessage(c)) = ev.original_content()
```

Comment:
Can you use a `match` statement so we get a single call to `ev.original_content()`?

Also, do you want to keep reactions too?

Reply:
Reactions weren't planned for the pined events feature, but I can try to add them too.

Reply:
Added reactions and polls.